### ISSUE TITLE
native ubuntu build of pytorch with cuda9/cudnn7/nccl2 fails to recognize V100

### ISSUE THREAD

**Issue by David-Levinthal** [pytorch/pytorch#3865]
I am attempting to benchmark fairseq-py across a spectrum of Nvidia cards. It appears to work with P4 and P40, but I get the following error when I set CUDA_VISIBLE_DEVICES to point to the V100.
THCudaCheck FAIL file=/home/levinth/pytorch/aten/src/THC/THCTensorCopy.cu line=204 error=48 : no kernel image is available for execution on the device
this is running on ubuntu 16.04, the Nvidia packages were installed using dpkg and *.deb packages
pytorch was built from source

it (fairseq-py) seems to run fine on the P40
| data-bin/wmt_de_en train 4500966 examples
| data-bin/wmt_de_en valid 2997 examples
| using 1 GPUs (with max tokens per GPU = 4000 and max sentences per GPU = None)
| model lstm_luong_wmt_en_de, criterion CrossEntropyCriterion
| epoch 001:   1000 / 37483 loss=9.68 (10.49), wps=4632, wpb=3514, bsz=126, lr=0.25, clip=100%, gnorm=1.02492

I will check the P100 after running this for a while, but suspect there will be no issue.

**Comment by ngimel** [comment]
How do you build pytorch? By default it tries to find GPUs on your system and builds for their architectures, if v100 was not visible at the time of build, you will get this error. Alternatively, you can set TORCH_CUDA_ARCH_LIST env variable to include all the archtectures you need, regardless of whether they are visible at compile time.

**Comment by David-Levinthal** [comment]
AHH..I did the build when only a couple P4s were in the box. I did wonder
about the issue of detection.
what should I set this to?
export TORCH_CUDA_ARCH_LIST="All"
?

**Comment by ngimel** [comment]
"6.0 6.1 7.0" if you are not going to run on Maxwell's. "All" will compile forever, and produce enormous binaries.

**Comment by David-Levinthal** [comment]
so I need to invoke setup.py install again and rebuild pytorch (I am new to
using pytorch)
Do I need to do anything to clean up the current build?
or should I delete the pytorch directory and clone a new version and then
run setup.py install
?

**Comment by ngimel** [comment]
```
python setup.py clean
TORCH_CUDA_ARCH_LIST="6.0 6.1 7.0" python setup.py install
```

**Comment by David-Levinthal** [comment]
did the trick (I had to invoke python3 as I have python2.7 and python3 on
the system)
